web_src/js/index.js

```diff
@@ -2377,6 +2377,19 @@ $(document).ready(async () => {
     });
   });
 
+  $('.pinned-repo-btn').on('click', function (e) {
+    const $this = $(this);
+    const $redirect = $this.data('redirect');
+    e.preventDefault();
+    $.post($this.data('url'), {
+      _csrf: csrf,
+      name: $this.data('name'),
+      status: $this.data('status')
+    }).done(() => {
+      window.location.href = $redirect;
```

Comment:
Can you retrieve new URL be extracted from the POST response? I guess we would not need this `redirect` variable then too.

Reply:
I have add ``ctx.Redirect`` (it will set Location and 302 status), and it's usefull when using a form (“pinnde other repo” method use this way now), But using JQury to do post work, It not redirect. I don't know why...

Reply:
If using form directly, (the style before ca1564988bc3ed5bf1276d292ab6ffee6e3dde53) It will work also, but it's hard to simplify ui..

Reply:
> But using JQury to do post work, It not redirect. I don't know why...

Only navigation requests (submitting a form) will result in the browser redirecting. For Ajax-triggered ones (jQuery, fetch), you'd need to parse the response headers yourself, e.g.

````js
const res = await fetch(url, {...opts});
if (res.status >= 300 && res.status < 400 && res.headers.location) {
  window.location.href = res.headers.location;
} 
```

Reply:
@silverwind  How about this way? Thanks.
1356eb9716c1b077556bcf2378e47f8b721dd278

Reply:
Any kind of page reload/redirect is generally bad UX. What does the button do and can the DOM be modified to achieve the same result without a page reload?

Reply:
Hmm, I think reloading is necessary, because the other data on this page may be changed in another way. should reload to get the latest data from the server. Then profile page does not contain too much data (like diff page in pr), reload won't take too much time. Thanks.